### PR TITLE
filter: support tc police action

### PR DESCRIPTION
**Pull request by Cordius**
TC police is implemented partially in fwmark filter only currently and cannot be reused by other filters.
This patchset supports TC police action and allows it be used by other filter such as u32.

### CODE REVIEW

File: filter.go [hunk 1]
```diff
@@ -260,6 +260,40 @@ func NewSkbEditAction() *SkbEditAction {
 	}
 }
 
+type PoliceAction struct {
+	ActionAttrs
+	Rate            uint32 // in byte per second
```
Comment (paravmellanox) [inline filter.go:265]: struct flow_action_entry in kernel defines rate as u64 field rate_bytes_ps.

Comment (Cordius) [reply filter.go:265]: struct tc_ratespec I referred to in kernel defines rate as u32 field.
```c
struct tc_ratespec {
        unsigned char   cell_log;
        __u8            linklayer; /* lower 4 bits */
        unsigned short  overhead;
        short           cell_align;
        unsigned short  mpu;
        __u32           rate;
};
```
And the c library iproute2 also regards the rate field as u32 value.

So which struct should I refered to?

Comment (paravmellanox) [reply filter.go:265]: do you have equivalent iproute2 command example for this API that may help.

Comment (Cordius) [reply filter.go:265]: Well, I have test it with the following iproute2 command.
```bash
tc qdisc add dev eth0 ingress
tc filter add dev eth0 parent ffff: u32 match u32 0 0 action police rate 1000000kbit burst 100mbit mtu 64k drop
```
And the second patch I sent provides a test about this API.

**Comment by aboch** [comment]
pls fix conflicts

**Comment by aboch** [comment]
@Cordius 

Please fix the conflicts

**Comment by Cordius** [comment]
> @Cordius
> 
> Please fix the conflicts

Done

**Comment by Cordius** [comment]
> @Cordius
> 
> Please fix the conflicts

Done

**Comment by aboch** [comment]
Please squash your commits into one.

**Comment by Cordius** [comment]
> Please squash your commits into one.

Done

**Pull request merged**